File: тесты/bench03_prime_sieve.py
import time

def count_primes(limit):
    is_prime = [True] * (limit + 1)
    is_prime[0] = is_prime[1] = False
    p = 2
    while p * p <= limit:
        if is_prime[p]:
            i = p * p
            while i <= limit:
                is_prime[i] = False
                i += p
        p += 1
    count = 0
    p = 2
    while p <= limit:
        if is_prime[p]:
            count += 1
        p += 1
    return count

if __name__ == "__main__":
    t0 = time.perf_counter()
    ans = count_primes(200000)
    t1 = time.perf_counter()
    ms = (t1 - t0) * 1000.0
    print(f"RESULT:{ans}")
    print(f"TIME_MS:{ms:.2f}")

File: тесты/bench03_prime_sieve.cpp
#include <iostream>
#include <vector>
#include <chrono>

int count_primes(int limit) {
    std::vector<bool> is_prime(limit + 1, true);
    is_prime[0] = is_prime[1] = false;
    for (int p = 2; p * p <= limit; p++) {
        if (is_prime[p]) {
            for (int i = p * p; i <= limit; i += p) {
                is_prime[i] = false;
            }
        }
    }
    int count = 0;
    for (int p = 2; p <= limit; p++) {
        if (is_prime[p]) count++;
    }
    return count;
}

int main() {
    auto t0 = std::chrono::high_resolution_clock::now();
    int ans = count_primes(200000);
    auto t1 = std::chrono::high_resolution_clock::now();
    double ms = std::chrono::duration<double, std::milli>(t1 - t0).count();
    std::cout << "RESULT:" << ans << std::endl;
    std::cout << "TIME_MS:" << ms << std::endl;
    return 0;
}
